app/src/main/java/org/simple/clinic/storage/DaoWithUpsert.kt

```diff
@@ -0,0 +1,34 @@
+package org.simple.clinic.storage
+
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.Transaction
+import android.arch.persistence.room.Update
+
+/**
+ * Mimics sqlite's `upsert` conflict strategy because Room's grammar doesn't understand it.
+ */
+abstract class DaoWithUpsert<T> {
+
+  @Insert(onConflict = OnConflictStrategy.IGNORE)
+  abstract fun insert(record: List<T>): List<Long>
+
+  @Update(onConflict = OnConflictStrategy.FAIL)
```

Comment:
@vinaysshenoy @sanchita I decided to use `FAIL` as the conflict strategy, but I'm not very sure if it's a good choice. Thoughts?

Reply:
Yes, that's what I understand.

Reply:
I think it's alright for now then. This will be caught by the upsert test anyway.